xarray/core/dataset.py

```diff
@@ -5440,10 +5435,10 @@ def unstack(
             else:
                 dims = list(dim)
 
-            missing_dims = [d for d in dims if d not in self.dims]
+            missing_dims = set(dims) - set(self.dims)
```

Comment:
I did a quick test, and this is indeed faster. It's equivalent for very small lists — 3 in each — and then much faster as the list grows.